test/OutputTest/OutputTest.cs

```diff
@@ -0,0 +1,191 @@
+            {
+                configureNetwork = new ConfigureNetwork()
+                {
+                    interfacenetworkif = new List<Interfacenetworkif>()
+                }
+            };
+            Interfacenetworkif INI = new Interfacenetworkif()
+            {
+                Name = "TestName"
+            };
+            AC.configureNetwork.interfacenetworkif.Add(INI);
+            testreturnConfigList = new TestOutput().objectToList(AC);
+            Assert.NotEmpty(testreturnConfigList);
+        }
+        [Fact]
+        public void ObjectToListProxySetNotNull()
+        {
+            AC = new ACConfig()
+            {
+                configureviop = new Configureviop()
+                {
+                    proxyset = new List<Proxyset>()
+                }
+            };
+            Proxyset PS = new Proxyset()
+            {
+                proxyname = "testProxyname"
+            };
+            AC.configureviop.proxyset.Add(PS);
+            testreturnConfigList = new TestOutput().objectToList(AC);
+            Assert.NotEmpty(testreturnConfigList);
+        }
+        [Fact]
+        public void ObjectToListProxyIpNotNull()
+        {
+            AC = new ACConfig()
+            {
+                configureviop = new Configureviop()
+                {
+                    proxyip = new List<Proxyip>()
+                }
+            };
+
+            Proxyip PI = new Proxyip()
+            {
+                ip = "213"
+            };
+            AC.configureviop.proxyip.Add(PI);
+            testreturnConfigList = new TestOutput().TestObjectToList(AC);
+            Assert.NotEmpty(testreturnConfigList);
+        }
+
+        [Fact]
+        public void TestWriteOutputFromList()
+        {
+            List<string> AC = new List<string>()
+            {
+                "Line1",
+                "Line 2 with whitespaces   ",
+                "This is Line 3 as sentence"
+            };
+          MemoryStream s = new MemoryStream();
+          s =  new TestOutput().WriteTestOutput(AC, s);
+          Assert.NotEmpty(s.ToArray());
+          Assert.NotNull(s);
+          byte[] bytes = s.ToArray();
+            using (var stream = new MemoryStream(bytes))
+            using (var reader = new StreamReader(stream))
+            {
+                var collection = new List<string>();
+                string input;
+
+                while ((input = reader.ReadLine()) != null)
+                    collection.Add(input);
+
+                Assert.Equal(3, collection.Count);
+            }
+        }
+    }
+
+    public class TestOutput : Output
+    {
+        public IEnumerable<string> GetConfigStringFromTestObject(object input)
+        {
+            return GetConfigStringFromObject(input);
+        }
+
+        public List<string> TestObjectToList(ACConfig AC)
+        {
+            return objectToList(AC);
+        }
+        public MemoryStream WriteTestOutput(List<string> AC, dynamic s)
```

Comment:
das hier würde ich dann ändern in:

```suggestion
        public string WriteTestOutput(List<string> AC)
         { 
            var output = "";
            using(var stream = new MemoryStream()){
               writeOutput(AC, stream);
               // reset stream position to 0. Because the writer may leave the stream at the end.
               stream.Position = 0;
               output = StreamReader(stream).ReadToEnd();
            }
            return output;
         }
```